apps/extension/src/features/router.tsx

```diff
@@ -0,0 +1,48 @@
+import {
+  type ReactLocationOptions,
+  type Route,
+  createMemoryHistory
+} from "@tanstack/react-location"
+
+import { Calendar } from "~features/calendar"
+import { trpc } from "~trpc"
+
+import { SignIn } from "./auth"
+import Settings from "./settings"
+
+// Create a memory history
+export const reactLocationOptions: ReactLocationOptions = {
+  history: createMemoryHistory({
+    initialEntries: ["/authenticated"] // Pass your initial url
+  })
+}
+
+export const getRoutes: () => Route[] = () => {
+  const utils = trpc.useContext()
```

Comment:
@p6l-richard you were previously passing the utils through props, any idea why, is this function ever called outside the trpc context provider?

Reply:
I think it was because I had some issues with the trpc headers not always including the accessToken so wasnted to pass on the authentication as a prop.

It's fine as you wrote it though.